coordinator/persistence-dse-6.8/src/main/java/io/stargate/db/dse/impl/DsePersistence.java

```diff
@@ -564,8 +565,11 @@ private <T extends Result> CompletableFuture<T> executeRequest(
                       // There is only 2 types of response that can come out: either a
                       // ResultMessage (which itself can of different kind), or an ErrorMessage.
                       if (response instanceof ErrorMessage) {
-                        throw convertExceptionWithWarnings(
-                            (Throwable) ((ErrorMessage) response).error);
+                        PersistenceException pe =
+                            convertExceptionWithWarnings(
+                                (Throwable) ((ErrorMessage) response).error);
+                        pe.setTracingId(MessageHandler.getTracingId(response));
```

Comment:
Why do we need the MessageHandler ? Can we simply call response.getTracingId ?

Reply:
`getTracingId()` of `Message.Response` is package protected so I ended up creating a class in that package and access that method. It was not accessible outside the package.

Reply:
I guess that is necessary for now, but split packages will become problematic with later JDKs.
But then again if that's the only way to force access for now, need to do so and worry about problem later on
(we won't need to support C* pre-5, or DSE 6.8 in future)

Reply:
Ok, this is only needed with Cassandra-4.0 (and probably 4.1 for SG 2.1) -- 3.11/DSE-6.8 have method `public` so can be accessed directly. Will push changes.